perceval/utils/density_matrix.py

```diff
@@ -500,27 +503,27 @@ def _construct_loss_operators(self, mode: int, p: float):
                                                                       p**n_photon_loss)
         return operators
 
-    @dispatch(int, float)
-    def apply_loss(self, mode: int, prob: float):
+    def apply_loss(self, modes: Union[int, list], prob: float):
         """
         Apply a loss on some mode according to some probability of losing a photon
         Everything works like if the mode was connected to some virtual mode with a beam splitter of reflectivity prob
-        :param mode: the mode were you want to simulate a loss
+
+        :param modes: the mode were you want to simulate a loss
         :param prob: the probability to lose a photon
         """
+        self._apply_loss(modes, prob)
+
+    @dispatch(int, float)
```

Comment:
In order to get the docstring automatically retrieved by Sphinx, you might want to either:

-  try single dispatch (with `@functools.singledispatch`, see: https://docs.python.org/3/library/functools.html
- gather both signatures to only one, and if the user passes an int, just turn it to a single element list, in order to have only one working implementation of your algorithm (that would expect a List[int] input parameter)